configuration/parser.go

```diff
@@ -205,6 +206,25 @@ func (p *Parser) overwriteStruct(v reflect.Value, fullpath string, path []string
 	byUpperCase := make(map[string]int)
 	for i := 0; i < v.NumField(); i++ {
 		sf := v.Type().Field(i)
+
+		// For fields inlined in the YAML configuration file, the environment variables also need to be inlined
+		// Example struct tag for inlined field: `yaml:",inline"`
+		yamlTag := sf.Tag.Get("yaml")
+		split := strings.Split(yamlTag, ",")
+		// Skip the first entry, which is the field name
+		isInlined := slices.Contains(split[1:], "inline")
+
+		if isInlined && sf.Type.Kind() == reflect.Struct {
```

Comment:
Can probably use `strings.Cut` for this;

```suggestion
		_, yamlOpts, _ := strings.Cut(sf.Tag.Get("yaml"), ",")
		if yamlOpts == "inline" && sf.Type.Kind() == reflect.Struct {
```

Reply:
Yep, that should work indeed; it is theoretically possible for a struct field to have multiple flags, which is why I split it into a slice initially:
<img width="435" alt="image" src="https://github.com/user-attachments/assets/8feae38c-d73b-4129-9ccf-11939bdf60a9" />
(from https://pkg.go.dev/gopkg.in/yaml.v3#Marshal)

 However, it seems like the `inline` flag is only used on its own and the YAML library also doesn't have any other flags that work together with it, so this should work just fine.